models/ResourceService.cfc

```diff
@@ -342,29 +361,36 @@ component singleton accessors="true" {
 	 * @throws ResourceBundle.InvalidBundlePath if bundlePath is not found
 	 */
 	array function getRBKeys( required rbFile, rbLocale = "" ) {
+
 		var keys = arrayNew( 1 );
 
 		// default locale?
 		if ( NOT len( arguments.rbLocale ) ) {
 			arguments.rbLocale = variables.defaultLocale;
 		}
 
-		// read file
-		var fis = getResourceFileInputStream( "#arguments.rbFile#_#arguments.rbLocale#.properties" );
-		var rb  = createObject( "java", "java.util.PropertyResourceBundle" ).init( fis );
-
-		try {
-			// Get Keys
-			var rbKeys = rb.getKeys();
-			// Loop through Keys and get the elements.
-			while ( rbKeys.hasMoreElements() ) {
-				arrayAppend( keys, rbKeys.nextElement() );
+		if ( variables.ResourceType =="java" ){
+			// read file
+			var fis = getResourceFileInputStream( "#arguments.rbFile#_#arguments.rbLocale#.properties" );
+			var rb  = createObject( "java", "java.util.PropertyResourceBundle" ).init( fis );
+
+			try {
+				// Get Keys
+				var rbKeys = rb.getKeys();
+				// Loop through Keys and get the elements.
+				while ( rbKeys.hasMoreElements() ) {
+					arrayAppend( keys, rbKeys.nextElement() );
+				}
+			} finally {
+				fis.close();
 			}
-		} finally {
-			fis.close();
+			return keys;
+		} else {
+			var myResource= _flattenStruct( deserializeJSON( fileRead( "#arguments.rbFile#_#arguments.rbLocale#.json" ) ) );
```

Comment:
Why not call `_loadJsonSubBundle()` if you have that code there already?

Reply:
done